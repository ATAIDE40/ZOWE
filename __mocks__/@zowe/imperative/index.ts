/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

/**
 * Mocks all imperative classes specified under __mocks__/__imperative__
 * @returns mocked version of the @zowe/imperative module
 */
const __imperative__get = function () {
  const fs = require.requireActual("fs");
  const imperativePath = require("path").resolve(__dirname, "..", "..", "__imperative__");
  const extraClasses: any = {};
  fs.readdirSync(imperativePath).forEach((file: string) => {
    extraClasses[file.split(".")[0]] = require(`${imperativePath}/${file}`);
  });
  return {
    ...require.requireActual("@zowe/imperative"),
    // Note: Comment out the line below if you do not want to mock
    //       the classes in __mocks__/__imperative__/ for local testing purposes
    ...extraClasses
    // Note: You can still use Imperative.console.info("debug") in your test
  };
}

module.exports = __imperative__get();
