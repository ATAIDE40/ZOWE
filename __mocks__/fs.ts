/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

const __mocks__path = require("path");

let __mocks__fs = jest.genMockFromModule("fs") as any;
const __mocks__oldReadFileSync = require.requireActual("fs").readFileSync;

// This is a custom function that our tests can use during setup to specify
// what the files on the "mock" filesystem should look like when any of the
// `fs` APIs are used.
let __mocks__mockFiles = Object.create(null);
function __mocks__fs____setMockFiles(newMockFiles: { [key: string]: string }) {
    __mocks__mockFiles = Object.create(null);
    for (const file in newMockFiles) {
        const dir = __mocks__path.dirname(file);
        if (!__mocks__mockFiles[dir]) {
            __mocks__mockFiles[dir] = {};
        }
        __mocks__mockFiles[dir][__mocks__path.basename(file)] = newMockFiles[file];
    }
}

// A custom version of `readdirSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__readdirSync(filePath: string) {
    return __mocks__mockFiles[filePath] || {};
}

// A custom version of `existsSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__existsSync(filePath: string) {
    const fileContents = __mocks__fs__readdirSync(__mocks__path.dirname(filePath))[__mocks__path.basename(filePath)];
    if (typeof fileContents === "undefined") {
        return false;
    }
    return true;
}

// A custom version of `readFileSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__readFileSync(filePath: string, encoding?: string) {
    // Don't mock if yargs is trying to load a locale json file
    if (filePath.match(/node_modules.yargs/)) {
        return __mocks__oldReadFileSync(filePath, encoding);
    }

    if (!__mocks__fs__existsSync(filePath)) {
        throw new Error("File not found");
    }
    return __mocks__fs__readdirSync(__mocks__path.dirname(filePath))[__mocks__path.basename(filePath)];
}

// A custom version of `lstatSync` that reads from the special mocked out
// file list set via __setMockFiles
function __mocks__fs__lstatSync(filePath: string) {
    return {
        isFile: () => false
    };
}

__mocks__fs.__setMockFiles = __mocks__fs____setMockFiles;
__mocks__fs.existsSync = __mocks__fs__existsSync;
__mocks__fs.readFileSync = __mocks__fs__readFileSync;
__mocks__fs.readdirSync = __mocks__fs__readdirSync;
__mocks__fs.lstatSync = __mocks__fs__lstatSync;

module.exports = __mocks__fs;
