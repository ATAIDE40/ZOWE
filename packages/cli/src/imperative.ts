/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import * as path from "path";

import { IImperativeConfig } from "@zowe/imperative";
import { Constants } from "./Constants";
import { ZosFilesOptions } from "./zosfiles/ZosFiles.options";

import { ZosmfSession } from "@zowe/zosmf-for-zowe-sdk";
import { ZosTsoProfile } from "@zowe/zos-tso-for-zowe-sdk";
import { ZosUssProfile } from "@zowe/zos-uss-for-zowe-sdk";

const config: IImperativeConfig = {
    productDisplayName: Constants.DISPLAY_NAME,
    commandModuleGlobs: [
        "daemon/*.definition!(.d).*s",
        "provisioning/*.definition!(.d).*s",
        "workflows/*.definition!(.d).*s",
        "zosconsole/*.definition!(.d).*s",
        "zosfiles/*.definition!(.d).*s",
        "zosjobs/*.definition!(.d).*s",
        "zosmf/*.definition!(.d).*s",
        "zostso/*.definition!(.d).*s",
        "zosuss/*.definition!(.d).*s",
        "zoslogs/*.definition!(.d).*s"
    ],
    rootCommandDescription: Constants.DESCRIPTION,
    defaultHome: Constants.HOME_DIR,
    envVariablePrefix: Constants.ENV_PREFIX,
    webHelpLogoImgPath: path.join(__dirname, "..", "web-help-logo.png"),
    logging: {
        appLogging: {
            logFile: Constants.LOG_LOCATION
        }
    },
    apimlConnLookup: [
        {
            apiId: "ibm.zosmf",
            gatewayUrl: "api/v1",
            connProfType: "zosmf"
        }
    ],
    baseProfile: {
        type: "base",
        schema: {
            type: "object",
            title: "Base Profile",
            description: "Base profile that stores values shared by multiple service profiles",
            properties: {
                host: {
                    type: "string",
                    optionDefinition: Constants.BASE_OPTION_HOST,
                    includeInTemplate: true
                },
                port: {
                    type: "number",
                    optionDefinition: Constants.BASE_OPTION_PORT
                },
                user: {
                    type: "string",
                    secure: true,
                    optionDefinition: Constants.BASE_OPTION_USER,
                    includeInTemplate: true
                },
                password: {
                    type: "string",
                    secure: true,
                    optionDefinition: Constants.BASE_OPTION_PASSWORD,
                    includeInTemplate: true
                },
                rejectUnauthorized: {
                    type: "boolean",
                    optionDefinition: Constants.BASE_OPTION_REJECT_UNAUTHORIZED,
                    includeInTemplate: true
                },
                tokenType: {
                    type: "string",
                    optionDefinition: Constants.BASE_OPTION_TOKEN_TYPE
                },
                tokenValue: {
                    type: "string",
                    secure: true,
                    optionDefinition: Constants.BASE_OPTION_TOKEN_VALUE
                },
                certFile: {
                    type: "string",
                    optionDefinition: Constants.BASE_OPTION_CERT_FILE
                },
                certKeyFile: {
                    type: "string",
                    optionDefinition: Constants.BASE_OPTION_CERT_KEY_FILE
                // },
                // certFilePassphrase: {
                //     type: "string",
                //     secure: true,
                //     optionDefinition: Constants.BASE_OPTION_CERT_FILE_PASSPHRASE
                }
            },
            required: []
        },
        createProfileExamples: [
            {
                options: "base1 --host example.com --port 443 --user admin --password 123456",
                description: "Create a profile called 'base1' to connect to host example.com and port 443"
            },
            {
                options: "base2 --host example.com --user admin --password 123456 --reject-unauthorized false",
                description: "Create a profile called 'base2' to connect to host example.com (default port - 443) " +
                    "and allow self-signed certificates"
            },
            {
                options: "base3 --host example.com --port 1443",
                description: "Create a profile called 'base3' to connect to host example.com and port 1443, " +
                    " not specifying a username or password so they are not stored on disk; these will need to be specified on every command"
            },
            {
                options: "base4 --reject-unauthorized false",
                description: "Create a zosmf profile called 'base4' to connect to default port 443 and allow self-signed certificates, " +
                    "not specifying a username, password, or host so they are not stored on disk; these will need to be specified on every command"
            }
        ],
        updateProfileExamples: [
            {
                options: "base1 --user newuser --password newp4ss",
                description: "Update a base profile named 'base1' with a new username and password"
            }
        ],
        authConfig: [
            {
                serviceName: "apiml",
                handler: __dirname + "/auth/ApimlAuthHandler",
                login: {
                    summary: Constants.APIML_LOGIN_SUMMARY,
                    description: Constants.APIML_LOGIN_DESCRIPTION,
                    examples: [
                        Constants.APIML_LOGIN_EXAMPLE1,
                        Constants.APIML_LOGIN_EXAMPLE2
                    ],
                    options: [
                        Constants.BASE_OPTION_HOST,
                        Constants.BASE_OPTION_PORT,
                        Constants.BASE_OPTION_USER,
                        Constants.BASE_OPTION_PASSWORD,
                        Constants.BASE_OPTION_REJECT_UNAUTHORIZED,
                        Constants.BASE_OPTION_CERT_FILE,
                        Constants.BASE_OPTION_CERT_KEY_FILE
                    ]
                },
                logout: {
                    summary: Constants.APIML_LOGOUT_SUMMARY,
                    description: Constants.APIML_LOGOUT_DESCRIPTION,
                    examples: [
                        Constants.APIML_LOGOUT_EXAMPLE1,
                        Constants.APIML_LOGOUT_EXAMPLE2
                    ],
                    options: [
                        Constants.BASE_OPTION_HOST,
                        Constants.BASE_OPTION_PORT,
                        Constants.APIML_LOGOUT_OPTION_TOKEN_TYPE,
                        Constants.BASE_OPTION_TOKEN_VALUE,
                        Constants.BASE_OPTION_REJECT_UNAUTHORIZED
                    ]
                }
            }
        ]
    },
    authGroupConfig: {
        authGroup: {
            summary: Constants.AUTH_GROUP_SUMMARY,
            description: Constants.AUTH_GROUP_DESCRIPTION
        }
    },
    configAutoInitCommandConfig: {
        handler: __dirname + "/config/auto-init/ApimlAutoInitHandler",
        provider: "APIML",
        autoInit: {
            options: [
                Constants.AUTO_INIT_OPTION_HOST,
                Constants.AUTO_INIT_OPTION_PORT,
                Constants.AUTO_INIT_OPTION_USER,
                Constants.AUTO_INIT_OPTION_PASSWORD,
                Constants.AUTO_INIT_OPTION_REJECT_UNAUTHORIZED,
                Constants.AUTO_INIT_OPTION_TOKEN_TYPE,
                Constants.AUTO_INIT_OPTION_TOKEN_VALUE
            ]
        },
        profileType: "base"
    },
    profiles: [
        {
            type: "zosmf",
            schema: {
                type: "object",
                title: "z/OSMF Profile",
                description: "z/OSMF Profile",
                properties: {
                    host: {
                        type: "string",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_HOST
                    },
                    port: {
                        type: "number",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_PORT,
                        includeInTemplate: true
                    },
                    user: {
                        type: "string",
                        secure: true,
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_USER
                    },
                    password: {
                        type: "string",
                        secure: true,
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_PASSWORD
                    },
                    rejectUnauthorized: {
                        type: "boolean",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_REJECT_UNAUTHORIZED
                    },
                    certFile: {
                        type: "string",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_CERT_FILE
                    },
                    certKeyFile: {
                        type: "string",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_CERT_KEY_FILE
                    },
                    // certFilePassphrase: {
                    //     type: "string",
                    //     secure: true,
                    //     optionDefinition: ZosmfSession.ZOSMF_OPTION_CERT_FILE_PASSPHRASE
                    // },
                    basePath: {
                        type: "string",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_BASE_PATH
                    },
                    protocol: {
                        type: "string",
                        optionDefinition: ZosmfSession.ZOSMF_OPTION_PROTOCOL
                    },
                    encoding: {
                        type: "string",
                        optionDefinition: {
                            name: "encoding",
                            aliases: ["ec"],
                            description: "The encoding for download and upload of z/OS data set and USS files." +
                                " The default encoding if not specified is IBM-1047.",
                            type: "string"
                        }
                    },
                    responseTimeout: {
                        type: "number",
                        optionDefinition: ZosFilesOptions.responseTimeout
                    }
                },
                required: []
            },
            createProfileExamples: [
                {
                    options: "zos123 --host zos123 --port 1443 --user ibmuser --password myp4ss",
                    description: "Create a zosmf profile called 'zos123' to connect to z/OSMF at host zos123 and port 1443"
                },
                {
                    options: "zos124 --host zos124 --user ibmuser --password myp4ss --reject-unauthorized false",
                    description: "Create a zosmf profile called 'zos124' to connect to z/OSMF at the host zos124 (default port - 443) " +
                        "and allow self-signed certificates"
                },
                {
                    options: "zos125 --host zos125 --port 1443",
                    description: "Create a zosmf profile called 'zos125' to connect to z/OSMF at the host zos125 and port 1443, " +
                        " not specifying a username or password so they are not stored on disk; these will need to be specified on every command"
                },
                {
                    options: "zos126 --reject-unauthorized false",
                    description: "Create a zosmf profile called 'zos126' to connect to z/OSMF on the default port 443 and allow self-signed " +
                        "certificates, not specifying a username, password, or host so they are not stored on disk; these will need to be " +
                        "specified on every command"
                },
                {
                    options: "zosAPIML --host zosAPIML --port 2020 --user ibmuser --password myp4ss --reject-unauthorized false --base-path basePath",
                    description: "Create a zosmf profile called 'zos124' to connect to z/OSMF at the host zos124 (default port - 443) " +
                        "and allow self-signed certificates"
                }
            ],
            updateProfileExamples: [
                {
                    options: "zos123 --user newuser --password newp4ss",
                    description: "Update a zosmf profile named 'zos123' with a new username and password"
                }
            ]
        },
        ZosTsoProfile,
        ZosUssProfile
    ]
};
module.exports = config;
