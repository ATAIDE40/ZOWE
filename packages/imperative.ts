/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

// WIP Imperative version of Zowe CLI
import { IImperativeConfig } from "@brightside/imperative";
import { Constants } from "./Constants";

const config: IImperativeConfig = {
    productDisplayName: Constants.DISPLAY_NAME,
    commandModuleGlobs: ["**/cli/*.definition!(.d).*s"],
    rootCommandDescription: Constants.DESCRIPTION,
    defaultHome: Constants.HOME_DIR,
    envVariablePrefix: Constants.ENV_PREFIX,
    logging: {
        appLogging: {
            logFile: Constants.LOG_LOCATION
        },
    },
    profiles: [
        {
            type: "zosmf",
            schema: {
                type: "object",
                title: "z/OSMF Profile",
                description: "z/OSMF Profile",
                properties: {
                    host: {
                        type: "string",
                        optionDefinition: {
                            name: "host",
                            aliases: ["H"],
                            description: "The z/OSMF server host name.",
                            type: "string",
                            required: true,
                        },
                    },
                    port: {
                        type: "number",
                        optionDefinition: {
                            name: "port",
                            aliases: ["P"],
                            description: "The z/OSMF server port.",
                            type: "number",
                            defaultValue: 443,
                        },
                    },
                    user: {
                        type: "string",
                        secure: true,
                        optionDefinition: {
                            name: "user",
                            aliases: ["u"],
                            description: "Mainframe (z/OSMF) user name, which can be the same as your TSO login.",
                            type: "string",
                            implies: ["password"],
                        },
                    },
                    pass: {
                        type: "string",
                        secure: true,
                        optionDefinition: {
                            name: "password",
                            aliases: ["p"],
                            description: "Mainframe (z/OSMF) password, which can be the same as your TSO password.",
                            type: "string",
                            implies: ["user"],
                        },
                    },
                    rejectUnauthorized: {
                        type: "boolean",
                        optionDefinition: {
                            name: "reject-unauthorized",
                            aliases: ["ru"],
                            description: "Reject self-signed certificates.",
                            type: "boolean",
                            defaultValue: "true",
                        },
                    },
                },
                required: ["host"],
            },
            createProfileExamples: [
                {
                    options: "zos123 --host zos123 --port 1443 --user ibmuser --password myp4ss",
                    description: "Create a zosmf profile called 'zos123' to connect to z/OSMF at host zos123 and port 1443"
                },
                {
                    options: "zos124 --host zos124 --user ibmuser --password myp4ss --reject-unauthorized false",
                    description: "Create a zosmf profile called 'zos124' to connect to z/OSMF at the host zos124 (default port - 443) " +
                    "and allow self-signed certificates"
                }
            ]
        },
        {
            type: "tso",
            schema: {
                type: "object",
                title: "TSO Profile",
                description: "z/OS TSO/E User Profile",
                properties: {
                    account: {
                        type: "string",
                        optionDefinition: {
                            name: "account",
                            aliases: ["a"],
                            description: "Your z/OS TSO/E accounting information.",
                            type: "string",
                            required: true,
                        },
                    },
                    characterSet: {
                        type: "string",
                        optionDefinition: {
                            name: "character-set",
                            aliases: ["cs"],
                            description: "Character set for address space to convert messages and responses from UTF-8 to EBCDIC.",
                            type: "string",
                            defaultValue: "697",
                        },
                    },
                    codePage: {
                        type: "string",
                        optionDefinition: {
                            name: "code-page",
                            aliases: ["cp"],
                            description: "Codepage value for TSO/E address space to convert messages and responses from UTF-8 to EBCDIC.",
                            type: "string",
                            defaultValue: "1047",
                        },
                    },
                    columns: {
                        type: "number",
                        optionDefinition: {
                            name: "columns",
                            aliases: ["cols"],
                            description: "The number of columns on a screen.",
                            type: "number",
                            defaultValue: 80,
                        },
                    },
                    logonProcedure: {
                        type: "string",
                        optionDefinition: {
                            name: "logon-procedure",
                            aliases: ["l"],
                            description: "The logon procedure to use when creating TSO procedures on your behalf.",
                            type: "string",
                            defaultValue: "IZUFPROC",
                        },
                    },
                    regionSize: {
                        type: "number",
                        optionDefinition: {
                            name: "region-size",
                            aliases: ["rs"],
                            description: "Region size for the TSO/E address space.",
                            type: "number",
                            defaultValue: 4096,
                        },
                    },
                    rows: {
                        type: "number",
                        optionDefinition: {
                            name: "rows",
                            description: "The number of rows on a screen.",
                            type: "number",
                            defaultValue: 24,
                        },
                    },
                },
                required: ["account"],
            },
            createProfileExamples: [
                {
                    description: "Create a tso profile called 'myprof' with default settings and JES accounting information of 'IZUACCT'",
                    options: "myprof -a IZUACCT"
                },
                {
                    description: "Create a tso profile called 'largeregion' with a region size of 8192, a logon procedure of MYPROC, and " +
                    "JES accounting information of '1234'",
                    options: "largeregion -a 1234 --rs 8192"
                }
            ]
        },
    ]
};
module.exports = config;
