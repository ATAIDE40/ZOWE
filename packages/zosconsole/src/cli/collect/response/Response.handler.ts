/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ICommandHandler, IHandlerParameters, Session } from "@brightside/imperative";
import { CollectCommand, ICollectParms, IConsoleResponse } from "../../../../../zosconsole";

/**
 * Handle to collect a MVS console command response
 * @export
 * @class Handler
 * @implements {ICommandHandler}
 */
export default class Handler implements ICommandHandler {

    public async process(commandParameters: IHandlerParameters) {

        const profile = commandParameters.profiles.get("zosmf");
        const session = new Session({
            type: "basic",
            hostname: profile.host,
            port: profile.port,
            user: profile.user,
            password: profile.pass,
            base64EncodedAuth: profile.auth,
            rejectUnauthorized: profile.rejectUnauthorized,
        });
        const collectParms: ICollectParms = {
            commandResponseKey: commandParameters.arguments.responsekey,
            consoleName: commandParameters.arguments["console-name"],
        };

        const response: IConsoleResponse = await CollectCommand.collect(session, collectParms);

        // Print out the response
        commandParameters.response.console.log(response.commandResponse);

        // Return as an object when using --response-format-json
        commandParameters.response.data.setObj(response);

    }
}
