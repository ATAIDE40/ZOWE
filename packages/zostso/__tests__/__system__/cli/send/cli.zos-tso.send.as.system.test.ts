/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ITestEnvironment } from "../../../../../../__tests__/__src__/environment/doc/response/ITestEnvironment";
import { runCliScript } from "../../../../../../__tests__/__src__/TestUtils";
import * as fs from "fs";
import { StartTso, StopTso } from "../../../../";
import { ITestSystemSchema } from "../../../../../../__tests__/__src__/properties/ITestSystemSchema";
import { Session } from "@brightside/imperative";
import { TestProperties } from "../../../../../../__tests__/__src__/properties/TestProperties";
import { TestEnvironment } from "../../../../../../__tests__/__src__/environment/TestEnvironment";

// Test Environment populated in the beforeAll();
let TEST_ENVIRONMENT: ITestEnvironment;
let systemProps: TestProperties;
let defaultSystem: ITestSystemSchema;
let REAL_SESSION: Session;
let acc: string;

describe("zos-tso send as", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "zos_tso_send_as",
            tempProfileTypes: ["zosmf", "tso"]
        });

        systemProps = new TestProperties(TEST_ENVIRONMENT.systemTestProperties);
        defaultSystem = systemProps.getDefaultSystem();

        REAL_SESSION = new Session({
            user: defaultSystem.zosmf.user,
            password: defaultSystem.zosmf.pass,
            hostname: defaultSystem.zosmf.host,
            port: defaultSystem.zosmf.port,
            type: "basic",
            rejectUnauthorized: defaultSystem.zosmf.rejectUnauthorized
        });
        acc = defaultSystem.tso.account;
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should successfully send data = \"time\"", async () => {
        const key = (await StartTso.start(REAL_SESSION, acc)).servletKey;
        const regex = fs.readFileSync(__dirname + "/__regex__/address_space_response.regex").toString();
        const response = runCliScript(__dirname + "/__scripts__/as/address_space_success.sh", TEST_ENVIRONMENT, [key]);
        StopTso.stop(REAL_SESSION, key);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(new RegExp(regex, "g").test(response.stdout.toString())).toBe(true);
    });

    describe("without profiles", () => {

        // Create a separate test environment for no profiles
        let TEST_ENVIRONMENT_NO_PROF: ITestEnvironment;
        let DEFAULT_SYSTEM_PROPS: ITestSystemSchema;

        beforeAll(async () => {
            TEST_ENVIRONMENT_NO_PROF = await TestEnvironment.setUp({
                testName: "zos_tso_send_as_without_profiles"
            });

            const sysProps = new TestProperties(TEST_ENVIRONMENT_NO_PROF.systemTestProperties);
            DEFAULT_SYSTEM_PROPS = sysProps.getDefaultSystem();
        });

        afterAll(async () => {
            await TestEnvironment.cleanUp(TEST_ENVIRONMENT_NO_PROF);
        });

        it("should successfully send data = \"time\" without a profile", async () => {
            const regex = fs.readFileSync(__dirname + "/__regex__/address_space_response.regex").toString();
            const key = (await StartTso.start(REAL_SESSION, acc)).servletKey;
            const response = runCliScript(__dirname + "/__scripts__/as/address_space_fully_qualified.sh",
                TEST_ENVIRONMENT_NO_PROF,
                [
                    key,
                    DEFAULT_SYSTEM_PROPS.zosmf.host,
                    DEFAULT_SYSTEM_PROPS.zosmf.port,
                    DEFAULT_SYSTEM_PROPS.zosmf.user,
                    DEFAULT_SYSTEM_PROPS.zosmf.pass
                ]);
            StopTso.stop(REAL_SESSION, key);
            expect(response.stderr.toString()).toBe("");
            expect(response.status).toBe(0);
            expect(new RegExp(regex, "g").test(response.stdout.toString())).toBe(true);
        });
    });
});
