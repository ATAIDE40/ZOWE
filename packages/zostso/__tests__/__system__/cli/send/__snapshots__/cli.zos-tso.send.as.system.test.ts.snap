// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zos-tso send as should display the help 1`] = `
"
 COMMAND NAME
 ------------

   address-space | as

 DESCRIPTION
 -----------

   Send data to the TSO address space, from which you previously started and
   received a token (a.k.a 'servlet-key').

 USAGE
 -----

   zowe zos-tso send address-space <servletKey> [options]

 POSITIONAL ARGUMENTS
 --------------------

   servletKey		 (string)

      The servlet key from a previously started TSO address space.

 REQUIRED OPTIONS
 ----------------

   --data  (string)

      The data to which we want to send to the TSO address space represented by the
      servlet key.

 PROFILE OPTIONS
 ---------------

   --zosmf-profile  | --zosmf-p (string)

      The name of a (zosmf) profile to load for this command execution.

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce the command response as a JSON document

   --help  | -h (boolean)

      Display help text


 EXAMPLES
 --------

   - \\"Send the TIME TSO command to the TSO address space
   identified by IBMUSER-329-aafkaaoc\\":

      $ zowe zos-tso send address-space IBMUSER-329-aafkaaoc --data \\"TIME\\"

{
  \\"success\\": true,
  \\"message\\": \\"The help was constructed for command: address-space.\\",
  \\"stdout\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   address-space | as\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Send data to the TSO address space, from which you previously started and\\\\n   received a token (a.k.a 'servlet-key').\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe zos-tso send address-space <servletKey> [options]\\\\n\\\\n POSITIONAL ARGUMENTS\\\\n --------------------\\\\n\\\\n   servletKey\\\\t\\\\t (string)\\\\n\\\\n      The servlet key from a previously started TSO address space.\\\\n\\\\n REQUIRED OPTIONS\\\\n ----------------\\\\n\\\\n   --data  (string)\\\\n\\\\n      The data to which we want to send to the TSO address space represented by the\\\\n      servlet key.\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --zosmf-profile  | --zosmf-p (string)\\\\n\\\\n      The name of a (zosmf) profile to load for this command execution.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce the command response as a JSON document\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - \\\\\\"Send the TIME TSO command to the TSO address space\\\\n   identified by IBMUSER-329-aafkaaoc\\\\\\":\\\\n\\\\n      $ zowe zos-tso send address-space IBMUSER-329-aafkaaoc --data \\\\\\"TIME\\\\\\"\\\\n\\\\n\\",
  \\"stderr\\": \\"\\",
  \\"data\\": \\"\\\\n COMMAND NAME\\\\n ------------\\\\n\\\\n   address-space | as\\\\n\\\\n DESCRIPTION\\\\n -----------\\\\n\\\\n   Send data to the TSO address space, from which you previously started and\\\\n   received a token (a.k.a 'servlet-key').\\\\n\\\\n USAGE\\\\n -----\\\\n\\\\n   zowe zos-tso send address-space <servletKey> [options]\\\\n\\\\n POSITIONAL ARGUMENTS\\\\n --------------------\\\\n\\\\n   servletKey\\\\t\\\\t (string)\\\\n\\\\n      The servlet key from a previously started TSO address space.\\\\n\\\\n REQUIRED OPTIONS\\\\n ----------------\\\\n\\\\n   --data  (string)\\\\n\\\\n      The data to which we want to send to the TSO address space represented by the\\\\n      servlet key.\\\\n\\\\n PROFILE OPTIONS\\\\n ---------------\\\\n\\\\n   --zosmf-profile  | --zosmf-p (string)\\\\n\\\\n      The name of a (zosmf) profile to load for this command execution.\\\\n\\\\n GLOBAL OPTIONS\\\\n --------------\\\\n\\\\n   --response-format-json  | --rfj (boolean)\\\\n\\\\n      Produce the command response as a JSON document\\\\n\\\\n   --help  | -h (boolean)\\\\n\\\\n      Display help text\\\\n\\\\n\\\\n EXAMPLES\\\\n --------\\\\n\\\\n   - \\\\\\"Send the TIME TSO command to the TSO address space\\\\n   identified by IBMUSER-329-aafkaaoc\\\\\\":\\\\n\\\\n      $ zowe zos-tso send address-space IBMUSER-329-aafkaaoc --data \\\\\\"TIME\\\\\\"\\\\n\\\\n\\"
}"
`;
